### PR TITLE
CSEN-26 Event entities were implemented

### PR DESCRIPTION
**Pull request by MedvedevBW**

### CODE REVIEW

**Comment by MedvedevBW** [comment]
@placccebo

File: src/main/scala/com/bwsw/cloudstack/entities/events/CloudStackEvent.scala [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+import com.bwsw.cloudstack.entities.events.account.{AccountCreateEvent, AccountDeleteEvent}
+import com.bwsw.cloudstack.entities.events.user.UserCreateEvent
+import com.bwsw.cloudstack.entities.events.vm.{VirtualMachineCreateEvent, VirtualMachineDestroyEvent}
+
+/**
+  * A base structure of Apache CloudStack event.
+  */
+@JsonTypeInfo(use = JsonTypeInfo.Id.NAME, property = "event", defaultImpl = classOf[CloudStackEvent], visible = true, include = JsonTypeInfo.As.EXISTING_PROPERTY)
+@JsonSubTypes(Array(
+  new Type(value = classOf[UserCreateEvent], name = "USER.CREATE"),
```
Comment (placccebo) [inline src/main/scala/com/bwsw/cloudstack/entities/events/CloudStackEvent.scala]: use Constants.Events for each name

File: src/main/scala/com/bwsw/cloudstack/entities/events/CloudStackEvent.scala [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+  new Type(value = classOf[AccountCreateEvent], name = "ACCOUNT.CREATE"),
+  new Type(value = classOf[AccountDeleteEvent], name = "ACCOUNT.DELETE"),
+  new Type(value = classOf[VirtualMachineCreateEvent], name = "VM.CREATE"),
+  new Type(value = classOf[VirtualMachineDestroyEvent], name = "VM.DESTROY")
+))
+class CloudStackEvent(val status: Option[String],
+                      @JsonProperty("event") val action: Option[String],
```
Comment (placccebo) [inline src/main/scala/com/bwsw/cloudstack/entities/events/CloudStackEvent.scala]: reverse the order of "event" field and "entityuuid" field

File: src/main/scala/com/bwsw/cloudstack/entities/events/vm/VirtualMachineCreateEvent.scala [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+* under the License.
+*/
+package com.bwsw.cloudstack.entities.events.vm
+
+import java.util.UUID
+
+import com.bwsw.cloudstack.entities.events.CloudStackEvent
+import com.bwsw.cloudstack.entities.events.Constants.Events
+
+case class VirtualMachineCreateEvent(override val status: Option[String], override val entityuuid: Option[UUID])
```
Comment (placccebo) [inline src/main/scala/com/bwsw/cloudstack/entities/events/vm/VirtualMachineCreateEvent.scala:26]: or Deploy?

File: src/test/scala/com/bwsw/cloudstack/entities/events/CloudStackEventTestSuite.scala [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+import com.bwsw.cloudstack.entities.events.Constants._
+import com.bwsw.cloudstack.entities.events.account.{AccountCreateEvent, AccountDeleteEvent}
+import com.bwsw.cloudstack.entities.events.user.UserCreateEvent
+import com.bwsw.cloudstack.entities.events.vm.{VirtualMachineCreateEvent, VirtualMachineDestroyEvent}
+import org.scalatest.FlatSpec
+
+class CloudStackEventTestSuite extends FlatSpec {
+  val status = Statuses.COMPLETE
+
+  it should "be deserialized to the UserCreateEvent if event is USER.CREATE" in {
```
Comment (placccebo) [inline src/test/scala/com/bwsw/cloudstack/entities/events/CloudStackEventTestSuite.scala]: use Constants.Events instead of "USER.CREATE" string

**Review by placccebo — CHANGES_REQUESTED** [review]

File: src/main/scala/com/bwsw/cloudstack/entities/events/CloudStackEvent.scala (hunk 1 above)
Comment (MedvedevBW) [reply src/main/scala/com/bwsw/cloudstack/entities/events/CloudStackEvent.scala]: ok, but compiler allows to use only "final" variable for it.

File: src/main/scala/com/bwsw/cloudstack/entities/events/vm/VirtualMachineCreateEvent.scala (hunk 3 above)
Comment (MedvedevBW) [reply src/main/scala/com/bwsw/cloudstack/entities/events/vm/VirtualMachineCreateEvent.scala:26]: No, the event json contains "VM.CREATE".

File: src/main/scala/com/bwsw/cloudstack/entities/events/CloudStackEvent.scala (hunk 2 above)
Comment (MedvedevBW) [reply src/main/scala/com/bwsw/cloudstack/entities/events/CloudStackEvent.scala]: ok

File: src/test/scala/com/bwsw/cloudstack/entities/events/CloudStackEventTestSuite.scala (hunk 4 above)
Comment (MedvedevBW) [reply src/test/scala/com/bwsw/cloudstack/entities/events/CloudStackEventTestSuite.scala]: ok

**Comment by MedvedevBW** [comment]
@placccebo

File: src/main/scala/com/bwsw/cloudstack/entities/events/vm/VirtualMachineCreateEvent.scala (hunk 3 above)
Comment (placccebo) [reply src/main/scala/com/bwsw/cloudstack/entities/events/vm/VirtualMachineCreateEvent.scala:26]: ok

**Review by placccebo — APPROVED** [review]

**Pull request merged**